tests/test_resampler.py

```diff
@@ -233,3 +240,53 @@ def test_resampling_function_init() -> None:
     assert ResamplingFunction(3) == ResamplingFunction.Min
     assert ResamplingFunction(4) == ResamplingFunction.Last
     assert ResamplingFunction(5) == ResamplingFunction.Count
+
+
+def test_resampler_first_timestamp() -> None:
+    """Test the resampler with the first timestamp."""
+    start = dt.datetime(1970, 1, 1, tzinfo=dt.timezone.utc)
+    step = dt.timedelta(seconds=0.5)
+    resampler = Resampler(
+        dt.timedelta(seconds=5),
+        ResamplingFunction.Average,
+        max_age_in_intervals=1,
+        start=start,
+        first_timestamp=True,
+    )
+
+    for i in range(20):
+        resampler.push_sample(timestamp=start + i * step, value=i + 1)
+
+    expected = [
+        (start + 0 * step, 5.5),
+        (start + 10 * step, 15.5),
+    ]
+
+    resampled = resampler.resample(start + 20 * step)
+
+    assert resampled == expected
+
+
+def test_resampler_last_timestamp() -> None:
+    """Test the resampler with the last timestamp."""
+    start = dt.datetime(1970, 1, 1, tzinfo=dt.timezone.utc)
+    step = dt.timedelta(seconds=0.5)
+    resampler = Resampler(
+        dt.timedelta(seconds=5),
+        ResamplingFunction.Average,
+        max_age_in_intervals=1,
+        start=start,
+        first_timestamp=False,
+    )
+
+    for i in range(20):
+        resampler.push_sample(timestamp=start + i * step, value=i + 1)
+
+    expected = [
+        (start + 10 * step, 5.5),
```

Comment:
The first bin is all samples in `[start, start + 9 * step]` but the label is `start + 10 * step`, right? So the bin label is the left timestamp in a left open bin (i.e part of the next bin). I find this definition confusing, is this what was used in the SDK?

Reply:
No, it's `[start, start + 10 * step]`: `1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 = 55`, `55 / 10 = 5.5`.

Reply:
That would be 11 timestamps (`start, start + 1 * step, ... start + 10 * step`), but 10 values.

Reply:
Oh, you are right, it is only 10 timestamps: `[start, start + 9 * step]` but the label is `start + 10 * step`. When building the buffer, we check for all `timestamps < buffer_start + interval`. `start + 10 * step` is part of the next resample. Let me check the SDK way. (@cwasicki would you prefer an inclusive interval?)

Reply:
The SDK uses inclusive behavior, as the bisect function checks `elem <= x`: https://github.com/frequenz-floss/frequenz-sdk-python/blob/v1.x.x/src/frequenz/sdk/timeseries/_resampling.py#L733

Will change that then.

Reply:
@cwasicki if `first_timestamp=True`, should it be left inclusive then? So the other way around?

Reply:
IMO the timestamp should always be the inclusive bin edge, which gives two options for a resampled timestamp $t_i$, which is the timestamp for all values at time $t$ in the bin with
* left-closed: $t_{i-1} < t \le t_i$
* right-closed: $t_i \le t < t_{i+1}$